### PR TITLE
Split from #42

### PR DESCRIPTION
**Pull request by pinkiebell**
EVM refactor split from #42 

*Attention*: accounts and accountsCode is commented out for now
because of stack depth problems if we use the `EthereumRuntime` in other contracts.

### CODE REVIEW

File: contracts/IEthereumRuntime.sol [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
-import "./Hash.slb";
 
 
 contract IEthereumRuntime {
-    using Hash for uint256[];
 
-    struct Result {
-        uint errno;
-        uint errpc;
+    struct EVMPreimage {
```
Comment (johannbarbie) [inline contracts/IEthereumRuntime.sol]: nice naming

File: test/ethereumRuntime.js [hunk 2]
```diff
@@ -56,32 +69,23 @@ contract('Runtime', function () {
...
+        const res = await rt.execute({ code, data, pc, gasLimit, stack, mem, accounts, accountsCode, logHash });
 
         if (fixture.result.stack) {
           assert.deepEqual(toStr(res.stack), fixture.result.stack);
         }
         if (fixture.result.memory) {
           assert.deepEqual(res.mem, fixture.result.memory);
         }
-        if (fixture.result.accounts) {
+        // TODO: re-enable once we support accounts again
```
Comment (johannbarbie) [inline test/ethereumRuntime.js]: what are these accounts exactly? touched accounts? like contracts that have been called?

File: test/helpers/runtimeAdapter.js [hunk 3]
```diff
@@ -5,36 +5,24 @@ export default class RuntimeAdapter {
...
-        ...this.buildArgs(code, data, params, stack, memory, accounts, accountsCode, logHash)
-      );
-  };
-
-  execute (code, data, gasLimit = BLOCK_GAS_LIMIT) {
+  execute ({ code, data, pc, stepCount, gasRemaining, gasLimit, stack, mem, accounts, accountsCode, logHash }) {
     return this.runtimeContract
       .execute(
-        ...this.buildArgs(code, data, [0, 0, BLOCK_GAS_LIMIT, gasLimit])
+        {
```
Comment (johannbarbie) [inline test/helpers/runtimeAdapter.js]: nice and clean 🎉

File: test/ethereumRuntime.js (hunk 2 above)
Comment (pinkiebell) [reply test/ethereumRuntime.js]: Touched/changed accounts. I will re-enable them once I resolve the gas and stack depth issue :)

**Pull request merged**